File: spring-boot/converter/src/main/java/com/example/converter/converter/AES256Converter.java
package com.example.converter.converter;

import com.example.converter.utils.AES256Util;
import lombok.RequiredArgsConstructor;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

@RequiredArgsConstructor
@Converter
public class AES256Converter implements AttributeConverter<String, String> {

    private final AES256Util aes256Util;

    @Override
    public String convertToDatabaseColumn(String attribute) {
        try {
            return aes256Util.encrypt(attribute);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

    @Override
    public String convertToEntityAttribute(String dbData) {
        try {
            return aes256Util.decrypt(dbData);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
}

File: spring-boot-kotlin/converter/src/main/kotlin/com/example/converter/member/converter/AES256Converter.kt
package com.example.converter.member.converter

import com.example.converter.utils.AES256Util
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter
class AES256Converter(
    private val aes256Util: AES256Util
) : AttributeConverter<String, String>{
    override fun convertToDatabaseColumn(attribute: String?): String {
        if (attribute == null) {
            return ""
        }
        return aes256Util.encrypt(attribute)
    }

    override fun convertToEntityAttribute(dbData: String?): String {
        if (dbData == null) {
            return ""
        }
        return aes256Util.decrypt(dbData);
    }
}